packages/nimble-components/src/table/specs/table-column-sort-hld.md

```diff
@@ -0,0 +1,75 @@
+
+For performance reasons, the table will not support custom sort functions on a column. A column can only be sorted based on a single field within the table's data, which guards against performance degradations from inefficient custom sort functions.
+
+#### Summary of new attributes on the base table column
+
+| attribute name | type                                                                                                         | default value                   | description                                                                                                                 |
+| -------------- | ------------------------------------------------------------------------------------------------------------ | ------------------------------- | --------------------------------------------------------------------------------------------------------------------------- |
+| sort-direction | `TableColumnSortDirection`, defined as `{none: undefined, ascending: 'ascending', descending: 'descending'}` | `TableColumnSortDirection.none` | The direction the column is sorted                                                                                          |
+| sort-index     | `number` or `null` \*                                                                                        | `null`                          | The index for sorting the column. When multiple columns are sorted, they will be sorted from lowest index to highest index. |
+| hidden         | `boolean`                                                                                                    | `false`                         | When set to true, do not render the column as part of the table                                                             |
+
+\* Note: The `sort-index` attribute is `number | null` because of the plan to use the `nullableNumberConverter` provided by FAST. That converter uses the value of `null` to represent non-number types rather than `undefined`, which is common within the nimble repo.
+
+### Validation
+
+The table will be invalid if it contains two columns with the same `sort-index`. If a duplicate `sort-index` is detected, the rows in the table will not render, no sort indicators will render, and the table's validity object will have `duplicateColumnSortIndex` set to `true`.
+
+### Visualizing the sort state
+
+Each column that is sorted will have an appropriate icon displayed in the header as shown in the design spec.
```

Comment:
Do we have clarity on whether the icon will appear on all columns that contribute to the sort or just the primary one? I couldn't tell from a quick scan of the design spec.

Reply:
Leslie's current guidance is that the icon will appear on all columns. I will clarify this sentence in the spec.